src/NSwag.CodeGeneration.TypeScript/Templates/Client.ProcessResponse.HandleStatusCode.liquid

```diff
@@ -37,12 +37,15 @@ let result{{ response.StatusCode }}: any = null;
 let resultData{{ response.StatusCode }}  = _responseText;
 {%             if response.UseDtoClass -%}
 {{ response.DataConversionCode }}
+{%             elseif operation.HandleReferences -%}
+result{{ response.StatusCode }} = _responseText === "" ? null : <{{ response.Type }}>jsonParse(resultData{{ response.StatusCode }}, this.jsonParseReviver);
 {%             else -%}
-result{{ response.StatusCode }} = JSON.parse(resultData{{ response.StatusCode }});
+try { result{{ response.StatusCode }} = (_responseText === "" ? null : <{{ response.Type }}>JSON.parse(resultData{{ response.StatusCode }}, this.jsonParseReviver)); } 
+catch(err) { result{{ response.StatusCode }} = _responseText; }
```

Comment:
Not sure whether this is a good idea here.
The generated code should not require external transformations in general (leaky abstraction) and should work out-of-the-box and thus deserialization should work as expected and a JSON exception should be thrown, no?

Reply:
In my suggested fix (see https://github.com/RicoSuter/NSwag/pull/2966/files), I just removed `JSON.parse(...)`. It seems that it is not needed at all in the context of axios if I am not wrong. See explanation here: https://github.com/RicoSuter/NSwag/pull/2966#issue-457198800 🤔